docs/userguide/format_parse/messages/mf2.md

```diff
@@ -0,0 +1,277 @@
+
+MessageFormat 2 will support more complex grammatical features, such as gender, inflections, and tagging parts of the message for style changes or speech.
+
+The reasoning for this effort is shared in the [“Why MessageFormat needs a successor”](https://github.com/unicode-org/message-format-wg/blob/main/docs/why_mf_next.md) document.
+
+The “MessageFormat 2” project, which develops the new data model, semantics, and syntax, is hosted on [GitHub](https://github.com/unicode-org/message-format-wg).
+
+The current specification for the syntax and data model can be found [here](https://github.com/unicode-org/message-format-wg/blob/main/spec/syntax.md).
+
+This tech preview implements enough of the MessageFormat functions to be useful, but the final set of functions and the parameters accepted by those functions is not yet finalized.
+
+
+
+## Examples
+
+### Basic usage
+
+```java
+import static org.junit.Assert.assertEquals;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import com.ibm.icu.message2.MessageFormatter;
+
+@Test
+public void testMf2() {
+    Map<String, Object> arguments = new HashMap<>();
+    arguments.put("name", "John"); // March 27, 2023, 7:42:51 PM
+    arguments.put("exp", new Date(1679971371000L));
+
+    MessageFormatter mf2 = MessageFormatter.builder()
+        .setPattern("{Hello {$name}, your card expires on {$exp :datetime skeleton=yMMMdE}!}")
+        .setLocale(Locale.UK)
+        .build();
+
+    assertEquals(
+        "Hello John, your card expires on Mon, 27 Mar 2023!",
+        mf2.formatToString(arguments));
+}
+```
+
+### Placeholder examples
+
+| Code to set runtime value for placeholder          | Examples of placeholder in message pattern                                   |
+|----------------------------------------------------|------------------------------------------------------------------------------|
+| `arguments.put("name", "John")`                    | `{$name}`                                                                    |
+| `arguments.put("exp", new Date(…))`                | `{$exp :datetime skeleton=yMMMdE}` <br/> `{$exp :datetime datestyle=full}!}` |
+| `arguments.put("val", 3.141592653)`                | `{$val}` <br/> `{$val :number skeleton=(.####)}`                             |
+| No argument, for fixed values known at build time  | `{(123456789.531) :number}`                                                  |
+
+
+### Plural selection message
+
+```java
+@Test
+public void testMf2Selection() {
+   String message = "match {$count :plural}\n"
+           + " when 1 {You have one notification.}\n"
+           + " when * {You have {$count} notifications.}\n";
+   Map<String, Object> arguments = new HashMap<>();
+
+
+   MessageFormatter mf2 = MessageFormatter.builder()
+       .setPattern(message)
+       .setLocale(Locale.UK)
```

Comment:
Define the locale like you define `message` (as a variable)

This isn't much of a test :-). We should start to think about the loop-over-locales tests...